src/Fmt_ast.ml

```diff
@@ -3273,35 +3308,63 @@ and fmt_module_declaration c ctx ~rec_flag ~first pmd =
       Sugar.functor_type c.cmts ~for_functor_kw:false
         (sub_mty ~ctx pmd_type)
   in
-  let colon =
-    match xmty.ast.pmty_desc with Pmty_alias _ -> false | _ -> true
+  let eqty =
+    match xmty.ast.pmty_desc with Pmty_alias _ -> None | _ -> Some ":"
   in
   Cmts.fmt c pmd_loc
-    (fmt_module c keyword pmd_name xargs None colon (Some xmty)
+    (fmt_module c keyword pmd_name xargs None ?eqty (Some xmty)
        pmd_attributes)
 
+and fmt_module_substitution c ctx pms =
+  let {pms_name; pms_manifest; pms_attributes; pms_loc} = pms in
+  update_config_maybe_disabled c pms_loc pms_attributes
+  @@ fun c ->
+  let xmty =
+    (* TODO: improve *)
```

Comment:
What needs to be improved here ?

Reply:
Sorry for the useless comment. The unsatisfying thing is the creation of a `module_type` from a `module_substitution`. The same commit also made changes to `fmt_module` that may deserve to be improved.